lib/chef/provider/package.rb

```diff
@@ -464,10 +464,7 @@ def current_version_array
 
       # @return [Array] new_version(s) as an array
       def new_version_array
-        @new_version_array ||=
-            [ new_resource.version ].flatten.map do |v|
-              ( v.nil? || v.empty? ) ? nil : v
-            end
+        [ new_resource.version ].flatten
```

Comment:
ah. That makes sense.

I'm cool with dropping the cache, I think. But we should add the `map` back... I think that's necessary.